### PR TITLE
Differentiate badge icon/color for indirect notifications

### PR DESCRIPTION
**Pull request by mahadevans87**
Currently Ferdi doesn't differentiate between direct and indirect notification badges
Co-Authored by <sampath3493@gmail.com>

### Description

 - Windows - Replace the icon used for showing indirect notifications in the taskbar to a blue(#0088cc) color (like slack, google chat etd)
  - All Platforms - Replace the red color used for indirect notifications in tabbed view for a service to 
#0088cc (Blue) color

### Motivation and Context
This PR is to address #583 

### How Has This Been Tested?
* This has been tested on Windows 10 on a Ferdi dev environment. Tray icon will not change to Blue in Mac / Linux as I have only replaced the overlayIcon for Windows. 

* We can call app.dock.setTrayIcon(${asset}) on Mac but this we would need a neat icon (png) to replace the existing badge. I thought this would be an overkill at this point. 

* I am not sure how we can change the color in Linux, so if a new feature request comes in, we can explore.

 
### Screenshots (if appropriate):
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project (run `$ yarn lint`).

### CODE REVIEW

**Comment by mahadevans87** [comment]
@vantezzen @kytwb - Can you guys please take a look?

**Comment by kytwb** [comment]
Thanks for the contribution! 🙏 
In case of a direct notification, is the badge red? with or without a count?

**Comment by mahadevans87** [comment]
> Thanks for the contribution! 🙏 
> In case of a direct notification, is the badge red? with or without a count?

In case of a direct message the badge stays red with the count

**Comment by kytwb** [comment]
On Windows, that would work, but on other platforms, would it affect the indirect messages dot color as well in the sidebar? If yes, that would mean the badge icons would not match anymore and might lead to confusion. In that case I recommend we merge that pull request once we replaced the icons for the other platforms as well. If not, then we can merge as a progressive enhancement for Windows only. What do you think?

**Comment by mahadevans87** [comment]
You are right. In Mac / Linux, merging this PR means that the dock badge wouldn't change its color for indirect notifications but the tab item badge would change its color to blue ( if indirect_notifications > 0 && direct_notifications == 0 ) for a particular service. 

The original enhancement request #583 was asked for Windows. What do you think? Should we hold off this PR or can we merge as a progressive enhancment for Windows only?

**Comment by kytwb** [comment]
I don't have a hard preference, but given it's not a high priority/blocking issue, I would suggest we take our time and try to collaborate on it so we get better at shipping cross-platform updates in the future. I am not sure what would be the best way for me to contribute for example to align macOS support. What do you think if I make a pull request to `Room4O4:feature/indirect_message_indicator`? So you can review the contribution, merge it and it should land in here, right?

**Comment by mahadevans87** [comment]
Yes that would be the best choice I feel. We can wait till you or someone else implements this for mac and linux and then we can merge to develop. Thank you

**Comment by FeikoJoosten** [comment]
I'm currently in the process of comitting my changes and setting up a pull-request. My version adds .ico and .png files for the tray icons and the taskbar. This includes Windows, Linux, macOS. It also adds support to switch the tray/taskbar icon to the correct indirect notification version. So it's basically @mahadevans87 version, but with a bit more logic on top.

**Comment by kytwb** [comment]
@all-contributors please add sampathBlam for review

**Comment by kytwb** [comment]
@all-contributors please add FeikoJoosten for code

**Comment by kytwb** [comment]
@mahadevans87 just to confirm, is this ready for review now?

**Comment by mahadevans87** [comment]
> 
> 
> @mahadevans87 just to confirm, is this ready for review now?

Yes. Thanks @FeikoJoosten

**Comment by kytwb** [comment]
On macOS, all my badges are blue now in the sidebar. 
Dock and tray icons show the red color properly.

Screen Shot 2020-04-22 at 12 35 30 PM

File: src/styles/tabs.scss [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
   .tab-item__message-count {
     align-items: center;
-    background: $theme-brand-danger;
+    background: #0088cc;
```
Comment (FeikoJoosten) [inline src/styles/tabs.scss:55]: @kytwb here's your issue. This change got merged in. This shouldn't be changed and should stay the same.

Comment (kytwb) [reply src/styles/tabs.scss:55]: Thank you, addressing.

**Pull request merged**